x-pack/legacy/plugins/siem/index.ts

```diff
@@ -151,27 +148,20 @@ export const siem = (kibana: any) => {
       mappings: savedObjectMappings,
     },
     init(server: Server) {
-      const { config, newPlatform, plugins, route } = server;
-      const { coreContext, env, setup } = newPlatform;
-      const initializerContext = { ...coreContext, env } as PluginInitializerContext;
-      const serverFacade = {
-        config,
-        usingEphemeralEncryptionKey:
-          get('usingEphemeralEncryptionKey', newPlatform.setup.plugins.encryptedSavedObjects) ??
-          false,
-        plugins: {
-          alerting: plugins.alerting,
-          actions: newPlatform.start.plugins.actions,
-          elasticsearch: plugins.elasticsearch,
-          spaces: plugins.spaces,
-          savedObjects: server.savedObjects.SavedObjectsClient,
-        },
-        route: route.bind(server),
+      const { coreContext, env, setup, start } = server.newPlatform;
+      const initializerContext = { ...coreContext, env };
+      const __legacy = {
+        config: server.config,
+        alerting: server.plugins.alerting,
+        route: server.route.bind(server),
       };
-      // @ts-ignore-next-line: setup.plugins is too loosely typed
-      plugin(initializerContext).setup(setup.core, setup.plugins);
 
-      initServerWithKibana(initializerContext, serverFacade);
+      // @ts-ignore-next-line: NewPlatform shim is too loosely typed
+      const pluginInstance = plugin(initializerContext);
+      // @ts-ignore-next-line: NewPlatform shim is too loosely typed
+      pluginInstance.setup(setup.core, setup.plugins, __legacy);
+      // @ts-ignore-next-line: NewPlatform shim is too loosely typed
```

Comment:
ouch on having to do ts-ignore-next-line. If we have to, we have to. Just some ouch. Any open tickets or other information about when and how these can be removed or is this is just temporary for a shim that will be altogether removed once we are out of the legacy folder?

Reply:
This is specific to the shim; this entire file will be removed on new platform.

Reply:
A few of these (CoreStart, CoreSetup) are properly typed, but because of the dynamic DI the plugins types are understandably loose. We could make this better, but we'd still have to cast parts of it, and this is all temporary code.